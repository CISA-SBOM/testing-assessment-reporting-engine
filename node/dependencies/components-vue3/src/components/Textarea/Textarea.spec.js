import { describe, expect, it } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Component from "./Textarea.vue";

describe("Textarea.vue", () => {
  it("is a Vue instance", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.vm).toBeTruthy();
  });

  it("matches snapshot with no props assigned", () => {
    const props = {};
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with maxlength and count-characters prop assigned", () => {
    const props = {
      maxlength: 50,
      countCharacters: true
    };
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with rows prop assigned", () => {
    const props = {
      rows: 2,
    };
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with disabled prop assigned", () => {
    const props = {
      disabled: true,
    };
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with readonly prop assigned", () => {
    const props = {
      readonly: true,
    };
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with placeholder prop assigned", () => {
    const props = {
      placeholder: "This is a placeholder",
    };
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with value prop assigned", () => {
    const props = {
      value: "This is the text content",
    };
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
