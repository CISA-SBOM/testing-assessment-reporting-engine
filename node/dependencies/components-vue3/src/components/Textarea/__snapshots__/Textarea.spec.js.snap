// Vitest Snapshot v1

exports[`Textarea.vue > matches snapshot with disabled prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea class=\\"form-control\\" rows=\\"5\\" maxlength=\\"524288\\" placeholder=\\"\\" disabled=\\"\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue > matches snapshot with maxlength and count-characters prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea class=\\"form-control\\" rows=\\"5\\" maxlength=\\"50\\" placeholder=\\"\\"></textarea>
  <div data-id=\\"sds-character-counter\\" class=\\"text-right text-gray-500\\">50</div>
</div>"
`;

exports[`Textarea.vue > matches snapshot with no props assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea class=\\"form-control\\" rows=\\"5\\" maxlength=\\"524288\\" placeholder=\\"\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue > matches snapshot with placeholder prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea class=\\"form-control\\" rows=\\"5\\" maxlength=\\"524288\\" placeholder=\\"This is a placeholder\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue > matches snapshot with readonly prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea class=\\"form-control\\" rows=\\"5\\" maxlength=\\"524288\\" placeholder=\\"\\" readonly=\\"\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue > matches snapshot with rows prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea class=\\"form-control\\" rows=\\"2\\" maxlength=\\"524288\\" placeholder=\\"\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue > matches snapshot with value prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\" value=\\"This is the text content\\"><textarea class=\\"form-control\\" rows=\\"5\\" maxlength=\\"524288\\" placeholder=\\"\\"></textarea>
  <!---->
</div>"
`;
